### PR TITLE
Add spawn options for sender and receiver in ssl application

### PR DESCRIPTION
**Pull request by zzydxm**
This change allows user to specify spawn options for tls_sender and
ssl_gen_statem processes in ssl config.
This could help on tuning trade-offs between CPU and Memory usage on
these processes.

### CODE REVIEW

File: lib/ssl/doc/src/ssl.xml [hunk 1]
```diff
@@ -919,6 +919,18 @@ fun(srp, Username :: binary(), UserState :: term()) ->
       </desc>
     </datatype>
 
+    <datatype>
+      <name name="spawn_opts"/>
+      <desc><p>Configures spawn options of TLS sender and receiver processes.</p>
+      <p>Setting up garbage collection options can be helpful on trade-offs between
```
Comment (RaimoNiskanen) [inline lib/ssl/doc/src/ssl.xml:925]: ... helpful *for* trade-offs ...

File: lib/ssl/doc/src/ssl.xml [hunk 2]
```diff
@@ -919,6 +919,18 @@ fun(srp, Username :: binary(), UserState :: term()) ->
...
+      CPU usage and Memory usage. 
+      See <seemfa marker="erts:erlang#spawn_opt/2"><c>erlang:spawn_opt/2</c></seemfa>.</p>
+      <p>For dist connections, default sender option is [{priority, max}], this priority
```
Comment (RaimoNiskanen) [inline lib/ssl/doc/src/ssl.xml:928]: <c>...</c> around `[{priority, max}]`

File: lib/ssl/doc/src/ssl.xml [hunk 3]
```diff
@@ -919,6 +919,18 @@ fun(srp, Username :: binary(), UserState :: term()) ->
...
+      option cannot be changed. For all connections, link is added to receiver and cannot
```
Comment (RaimoNiskanen) [inline lib/ssl/doc/src/ssl.xml:929]: <c>'...'</c> around `link`

**Comment by IngelaAndin** [comment]
@zzydxm  Could you please rebase on the latest master branch

**Comment by zzydxm** [comment]
> @zzydxm Could you please rebase on the latest master branch

Sorry I just saw the message, already updated the branch.

**Comment by u3s** [comment]
I've added a commit with adjustment so that spawn_opts_option test case does not fail after 744af235b8. Details in commit message.

**Comment by IngelaAndin** [comment]
@zzydxm Thank you for the PR it is now ready for inclusion.

**Pull request merged**